app/kubernetes-ingress-controller/2.1.x/guides/upgrade.md

```diff
@@ -0,0 +1,343 @@
+---
+title: Upgrading to 2.1.x
+---
+
+This guide walks through backwards incompatible changes in the Kong Kubernetes
+Ingress Controller (KIC) from to v2.1.x to help operators evaluate if any
+changes to their configuration are needed to upgrade, provides
+guidance on how to build testing environments to validate the upgrade, and
+walks through an upgrade of the Kubernetes Ingress Controller (KIC) using
+its [Helm Chart][chart].
+
+## Prerequisites
+
+* [Helm v3][helm] is installed
+* You are familiar with Helm `install` and `upgrade` operations. See the [documentation for Helm v3][helm-docs].
+
+> **Note:** Deploying and upgrading via Helm is the supported mechanism for
+production deployments of KIC. If you're deploying KIC using Kustomize or
+some other mechanism, you need to develop and test your own upgrade strategy
+based on the following examples.
+
+[helm]:https://helm.sh/
+[chart]:https://github.com/kong/charts
+[list-releases]:https://v3.helm.sh/docs/helm/helm_list/
+[helm-docs]:https://v3.helm.sh/docs/
+
+## Upgrading from 2.0.x to 2.1.x
+
+This document primarily covers the requirements for upgrading from 1.x to 2.x.
+If you have already upgraded to 2.0.x, only the steps in this subsection are
+necessary. If you are still on 1.x, follow the subsequent sections to upgrade
+to 2.0.x first. **You must upgrade to 2.0.x before upgrading to 2.1.x.**
+
+### Update CRDs
+
+The previous KongIngress CRD included several incorrectly-named fields which
+did not apply correctly to Kong upstreams. 2.1.x CRDs fix these fields:
+
+* `healthchecks.passive.unhealthy.timeout` is now
+  `healthchecks.passive.unhealthy.timeouts`
+* `healthchecks.active.unhealthy.timeout` is now
+  `healthchecks.active.unhealthy.timeouts`
+
+Review your KongIngresses to see if you use either of the old fields. You will
+need to re-add them with the new field name after upgrading. Kubernetes does
+not allow unknown fields in CRDs, and will strip the old field and its value
+after updating. There is no means of copying these into the new field
+automatically.
```

Comment:
```suggestion
Before upgrading, review your KongIngresses to see if you use either of the old 
fields. You will need to manually re-add them with the new field name after 
upgrading. 

Kubernetes does not allow unknown fields in CRDs. Since the old fields don't 
exist in 2.1, Kubernetes will strip the old field and its value after updating. 
There is no automated way to copy the old values into the new field.
```